azure_storage/azure_storage_api.py

```diff
@@ -234,3 +235,91 @@ async def get_directories(container_client):
     except FolderListError as error:
         print(error)
         return []
+
+async def get_pipeline_info(
+        connection_string: str,
+        pipeline_container_name: str,
+        pipeline_version: str
+    ) -> json:
+    """
+    Retrieves the pipeline information from Azure Blob Storage based on the provided parameters.
+
+    Args:
+        connection_string (str): The connection string for the Azure Blob Storage.
+        pipeline_container_name (str): The name of the container where the pipeline files are stored.
+        pipeline_version (str): The version of the pipeline to retrieve.
+
+    Returns:
+        json: The pipeline information in JSON format.
+
+    Raises:
+        PipelineNotFoundError: If the specified version of the pipeline is not found.
+
+    """
+    try:
+        blob_service_client = BlobServiceClient.from_connection_string(
+            connection_string
+        )
+
+        if blob_service_client:
+            container_client = blob_service_client.get_container_client(
+                pipeline_container_name
+            )
+            
+            blob_list = container_client.list_blobs()
+            for blob in blob_list:
+                if blob.name.split(".")[-1] != "json":
+                    print("WARNING a non JSON file is in the folder")
+                else:
+                    json_blob = await get_blob(container_client, blob.name)
+                    pipeline = json.loads(json_blob)
+                    if not isinstance(pipeline, list) and pipeline.get("version") == pipeline_version:
+                        return pipeline
+
+            else:
+                raise PipelineNotFoundError(
+                    "This version of the pipeline was not found."
+                )
+        
+    except PipelineNotFoundError as error:
+        print(error)
+        return False
+    
+    except FolderListError as error:
+        print(error)
+        return False
+
+def insert_new_version_pipeline(
+        pipelines_json: dict,
+        connection_string: str,
+        pipleine_container_name: str
+    ) -> bool:
+    """
+    Inserts a new version of a pipeline JSON into an Azure Blob Storage container.
+
+    Args:
+        pipelines_json (dict): The JSON data of the pipeline.
+        connection_string (str): The connection string for the Azure Blob Storage account.
+        pipleine_container_name (str): The name of the container where the pipeline JSON will be uploaded.
+
+    Returns:
+        bool: True if the pipeline JSON was successfully uploaded, False otherwise.
+    """
+    try:
+        blob_service_client = BlobServiceClient.from_connection_string(
+            connection_string
+        )
+
+        if blob_service_client:
+            container_client = blob_service_client.get_container_client(
+                pipleine_container_name
```

Comment:
*pipeline - only works right now because the function parameter also has the typo.